app/assets/javascripts/index/directions.js

```diff
@@ -82,17 +84,28 @@ OSM.Directions = function (map) {
     input.on("change", function (e) {
       // make text the same in both text boxes
       var value = e.target.value;
-      endpoint.setValue(value);
+      endpoint.setValue(value, false);
     });
 
-    endpoint.setValue = function (value, latlng) {
+    endpoint.setValue = function (value, force_reverse_geocoding, latlng) {
```

Comment:
What is `force_reverse_geocoding`? Is there any code here that actually does reverse geocoding?

Reply:
Yes, there is. Reverse geocoding is done when user right-clicks on map and selects "Directions from here" or "Directions to here".

Variable force_reverse_geocoding is introduced to pass information that user did this (right clicked and selected from context menu) and to perform reverse geocoding.

Reply:
There's no forcing of reverse geocoding. There's no code calling `reverse` like [here](https://github.com/openstreetmap/openstreetmap-website/blob/3debc151a86ed0016364d2d1348ecd82f5fa75fc/app/controllers/geocoder_controller.rb#L184). Nominatim will usually do reverse geocoding when "Directions from/to here" are selected, but that's because it decides on its own that the input query looks like coordinates. It won't make this decision correctly every time.

https://github.com/user-attachments/assets/4a12304d-e834-46f3-ae8f-4a6e147e5144

Reply:
Thanks for catching this! I've just updated with adding code for calling "reverse" like in above link.

Reply:
> Thanks for catching this! I've just updated with adding code for calling "reverse" like in above link.

We have just gotten another idea how to improve this part. I'll make another push soon.

Reply:
@AntonKhorev I've just pushed the latest fixes.

We improved logic of what is displayed in input boxes after right-clicking and selecting option from context menu. Now, if nominatim doesn't return anything or it returns something that is more than 200m away from clicked position, we will display coordinates (Number, Number), otherwise we will display what nominatim returned. The logic behind is "in input boxes should be displayed user-friendly text iff user entered it manually or user clicked on / or near the object recognized by nominatim, otherwise, coordinates will be displayed".

For calculating distance between position where user clicked and position of what nominatim returned, we used distanceTo method provided by Leaflet's LatLng which reduces it to calculating distance on great circle. After testing on Longyearbyen and Longyearbyen, it seems this approximation works fine.

Please, let me know if we can further improve it.

Thanks,
Nenad.